Sources/GRPC/ConnectionManager.swift

```diff
@@ -860,6 +891,62 @@ extension ConnectionManager {
   }
 }
 
+extension ConnectionManager {
+  /// Returns a synchronous view of the connection manager; each operation requires the caller to be
+  /// executing on the same `EventLoop` as the connection manager.
+  internal var sync: Sync {
+    return Sync(self)
+  }
+
+  internal struct Sync {
+    private let manager: ConnectionManager
+
+    fileprivate init(_ manager: ConnectionManager) {
+      self.manager = manager
+    }
+
+    /// A delegate for connectivity changes.
+    internal var connectivityDelegate: ConnectionManagerConnectivityDelegate? {
```

Comment:
Comment didn't post the first time :(

The `Delegate` pattern usually requires the delegate to be `weak` to avoid a reference cycle. I can't see/find the `connectivityDelegate` in `ConnectionManager`, but is it weak?

Reply:
Only applies if your delegates are classes of course

Reply:
It only requires a weak reference in cases where that cycle isn't broken manually due to some other lifecycle event. Generally in server-side code we prefer to do that (because weak is sloooooow).

Reply:
Ah fair enough! Ignore then, I'll resolve the comment :)